pkg/model/model.go

```diff
@@ -123,12 +124,12 @@ type Datapoint struct {
 	Timestamp *time.Time
 }
 
-type CloudwatchMetricResult struct {
-	Context *JobContext
-	Data    []*CloudwatchData
+type ScrapeResult[T CloudwatchData | TaggedResource] struct {
```

Comment:
A bit of nitpick: given that TaggedResource is only used for discovery jobs, might be more easily digestible to just have two separate structs here instead of using generics?

Reply:
Totally fair, I'll split this up to two separate structs